src/twisted/protocols/_smb/core.py

```diff
@@ -0,0 +1,760 @@
+            packet.close()
+            log.error("Unknown packet type")
+            log.debug("packet data {data!r}", data=packet.data[offset : offset + 64])
+            return
+        packet.hdr, o2 = _base.unpack(HeaderSync, packet.data, offset, _base.OFFSET)
+        isAsync = (packet.hdr.flags & FLAG_ASYNC) > 0
+        isRelated = (packet.hdr.flags & FLAG_RELATED) > 0
+        isSigned = (packet.hdr.flags & FLAG_SIGNED) > 0
+        # FIXME other flags 3.1 or too obscure
+        if isAsync:
+            packet.hdr = _base.unpack(HeaderAsync, packet.data, offset)
+        if isRelated:
+            this_packet = packet.data[offset : offset + packet.hdr.next_command]
+        else:
+            this_packet = packet.data[offset:]
+        flags_desc = ""
+        if isAsync:
+            flags_desc += " ASYNC"
+        if isRelated:
+            flags_desc += " RELATED"
+        if isSigned:
+            flags_desc += " SIGNED"
+        log.debug(
+            """
+HEADER
+------
+protocol ID     {pid!r}
+size            {hs}
+credit charge   {cc}
+status          {status}
+command         {cmd!r} {cmdn:02x}
+credit request  {cr}
+flags           0x{flags:04x}{flags_desc}
+next command    0x{nc:x}
+message ID      0x{mid:x}
+session ID      0x{sid:x}
+async ID        0x{aid:x}
+tree ID         0x{tid:x}
+signature       {sig}""",
+            pid=protocol_id,
+            hs=packet.hdr.size,
+            cc=packet.hdr.credit_charge,
+            status=packet.hdr.status,
+            cmd=COMMANDS[packet.hdr.command][0],
+            cmdn=packet.hdr.command,
+            cr=packet.hdr.credit_request,
+            flags=packet.hdr.flags,
+            flags_desc=flags_desc,
+            nc=packet.hdr.next_command,
+            mid=packet.hdr.message_id,
+            sid=packet.hdr.session_id,
+            aid=packet.hdr.async_id,
+            tid=packet.hdr.tree_id,
+            sig=binascii.hexlify(packet.hdr.signature),
+        )
+        if packet.hdr.command < len(COMMANDS):
+            name, req_type, resp_type = COMMANDS[packet.hdr.command]
+            func = "smb_" + name
+            try:
+                if func in globals() and req_type:
+                    req = _base.unpack(req_type, packet.data, o2)
+                    new_packet = packet.clone(
+                        data=this_packet, hdr=packet.hdr, body=req
+                    )
+                    globals()[func](new_packet, resp_type)
+                else:
+                    log.error(
+                        "command '{cmd}' not implemented",
+                        cmd=COMMANDS[packet.hdr.command][0],
+                    )
+                    errorResponse(packet, NTStatus.NOT_IMPLEMENTED)
+            except _base.SMBError as e:
+                log.error("SMB error: {e}", e=str(e))
+                errorResponse(packet, e.ntstatus)
+            except BaseException:
+                log.failure("in {cmd}", cmd=COMMANDS[packet.hdr.command][0])
```

Comment:
You already pulled this value out above, no need to go doing the dispatch lookup again.

```suggestion
                log.failure("in {cmd}", cmd=name)
```